webapp/channels/src/components/file_preview_modal/image_preview.tsx

```diff
@@ -27,22 +72,157 @@ export default function ImagePreview({fileInfo, canDownloadFiles}: Props) {
         previewUrl = fileInfo.has_preview_image ? getFilePreviewUrl(fileInfo.id) : fileUrl;
     }
 
+    const computeScaleOptions = React.useCallback(() => {
+        const imgEl = imageRef.current;
+
+        if (imgEl) {
+            setBaseScale(imgEl.naturalWidth / imgEl.clientWidth);
+            setScaleOptions([
+                {
+                    key: 'auto',
+                    value: 1,
+                    label: formatMessage({id: 'file_preview_modal_image_zoom_actions.select.auto', defaultMessage: 'Automatic zoom'}),
+                },
+                {
+                    key: 'actual',
+                    value: Math.round((imgEl.naturalWidth / imgEl.clientWidth) * 1000) / 1000, // round to 3 decimal places to prevent a duplicate value conflict with 100%
```

Comment:
Are you sure that this won't still conflict in certain cases? I know floating point equality in JS is weird, but it seems like these would still be duplicate values for square images. That said, the problem here might be more that we have two duplicate values in the list.

@abhijit-singh Is it intended that we have both "actual size" and "100%" in this list since they should be the same thing?

Reply:
Yes in theory there could be duplicates if the image was exactly the same size as the container div of the image, but it seemed very unlikely so I just left it. I also just wasn't really sure how to account for that, as `1 === 1.0` is true. Setting a custom value for 'Automatic Zoom' like 'auto' or 0 and then adding some conditional logic to change that to a 1 inside `selectZoomLevel` would probably be the simplest workaround. What do you think?

Reply:
Yeah, using a custom value for that makes sense to me. You'd have to add an if statement or two to `selectZoomLevel` to turn `auto` into `1`, but I think that's a straightforward approach. I'm sure that'll help the next person to work on this code to understand what's going on here

Reply:
So I think this will only work in the case where the user selects 'automatic zoom' from the select menu. If the user uses the mouse wheel or the plus / minus buttons to zoom then the select menu will show the zoom level, not 'automatic zoom'. What do you think?